build: TC builds only use vendored source code

jordanlewis opened the pull request:
`build/builder.sh` now mounts **only** the `github.com/cockroachdb/cockroach` tree under `$GOPATH/src` when a new environment variable, `BUILDER_HIDE_GOPATH_SRC`, is set to 1.

This option is now set in all TeamCity invocation scripts.

In order to make this work, we also vendor glide itself to prevent needing to `go get` it, which would fetch an outside dependency and spoil our supper.

As a consequence of the above, the segment of checkdeps that asserts there are no unvendored dependencies is rendered redundant, so it's deleted.

Updates #14451.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/14454)
<!-- Reviewable:end -->

jordanlewis committed:
build: TC builds only use vendored source code

`build/builder.sh` now mounts **only** the
`github.com/cockroachdb/cockroach` tree under `$GOPATH/src` when a
new environment variable, `BUILDER_HIDE_UNVENDORED`, is set to 1.

This option is now set in all TeamCity invocation scripts.

tamird commented on build/builder.sh:
```diff
@@ -110,7 +110,13 @@ mkdir -p "${host_home}"/.yarn-cache
-vols="${vols} --volume=${gopath0}/src:/go/src"
+if [ "${BUILDER_HIDE_UNVENDORED}" == "1" ]; then
+  # If BUILDER_HIDE_UNVENDORED is set to 1, mount just the cockroach source.
+  vols="${vols} --volume=${gopath0}/src/github.com/cockroachdb/cockroach:/go/src/github.com/cockroachdb/cockroach"
```
this is already done on L130, remove it?

jordanlewis replied:
Good point, also this was wrong. Fixed.

tamird reviewed (commented):
At the risk of being magic: https://github.com/cockroachdb/cockroach/issues/14451#issuecomment-290217397

dt commented:
totally ignorable bikeshedding nit: s/HIDE_UNVENDORED/ONLY_COCKROACH_SRC/ or something, since it isn't actually vendoring-specific.

jordanlewis commented:
I'm going to stick with this explicit approach. For instance, `build/teamcity-checkdeps.sh` doesn't use `make` as the command to run.

petermattis reviewed (approved):
LGTM

tamird commented on build/teamcity-check.sh:
```diff
@@ -1,12 +1,15 @@
 #!/usr/bin/env bash
 set -exuo pipefail
+
```
why the double newlines here and in acceptance? everywhere else only one is used.

jordanlewis replied:
Fixed

jordanlewis committed:
build: vendor glide

This is the only remaining build tool that wasn't available via vendor.

jordanlewis committed:
build: remove unvendored deps check

We no longer need to check that we don't have any unvendored
dependencies, because CI now doesn't mount any packages from the host
besides the cockroach source tree under test.

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 15 of 15 files at r3, 10 of 10 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[glide.yaml, line 20 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14454#-KgREhkvXCQ-wVRufxgX:-KgREhkvXCQ-wVRufxgY:b-3b1i6z) ([raw file](https://github.com/cockroachdb/cockroach/blob/bc266e360fc0934efc24b4fbb94c8fa528b548c6/glide.yaml#L20)):*
> ```YAML
>   version: 9d55a95b90eb10d08a07e0213d23797dbd7b7552
>   repo: https://github.com/andreimatei/grpc-go
> # We pin glide itself to prevent unexpected changes to glide's dependency
> ```

s/unexpected // (just the first one.

---

*[build/README.md, line 46 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/14454#-KgREw_7J1m2vgYXEzyq:-KgREw_7J1m2vgYXEzyr:b4n7ljd) ([raw file](https://github.com/cockroachdb/cockroach/blob/bc266e360fc0934efc24b4fbb94c8fa528b548c6/build/README.md#L46)):*
> ```Markdown
> This snapshot was built and is managed using `glide`.
> 
> ### Installing Glide
> ```

I would replace this section with a note in the previous section, though this is fine too if you don't feel like spending more time here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14454#-:-KgREnqrCIu0QR9yXE5N:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 13 of 13 files at r6, 10 of 10 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14454)*
<!-- Sent from Reviewable.io -->
